### DISCUSSION TITLE
Using the explorationTutorial and riseExploration

### DISCUSSION THREAD

**Discussion by dcbdan** [rise-lang/shine#209] [Q&A]
Ultimately, given a RISE expression similar to the one in explorationTutorial, I'd like to generate a candidate gpu and cpu kernel, potentially with a parameter that would control how long the search process takes. How does one do that?

Here is what I tried..
After running the riseExploration tutorial, the following Random directory was created:
```
Random/
├── Executor
│   ├── 178547dc
│   │   ├── 178547dc
│   │   ├── 178547dc_2759346c.c
│   │   ├── 178547dc_strategies
│   │   └── 2759346c
│   ...........
│   executor.csv
├── Expressions
│   ├── 178547dc
│   │   ├── 178547dc
│   │   └── strategies
│   ..........
├── Random_1630613798429.dot
├── Random_1630613798429_plain.dot
...
├── Random.csv
├── Random.dot
└── Random_plain.dot
```
where each of the *.c files contains a generated openMP kernel. 

Some questions:
- How should the executor.csv and Random.csv be used? For example, in executor.csv, a number of iterations yields a runtime of 5.3E-5 where as all the others are 15-20. Moreover, for at least one of the 5.3E-5 iterations, the provided kernel was the same as the naive "compute_gold" kernel.
- How do you specify openCL or CUDA output instead? If I'm reading it correctly, I think the riseExploration only  provides support for when the json's executor.name is "C" and not "openMP" or "openCL". (Which is confusing because the kernels in the *.c files had openMP pragmas despite the json specifying "C").

**Comment by michel-steuwer** [comment]
Hi @dcbdan,

thanks for your interest in RISE and our interesting chat this week.
The exploration you are trying out is a work in progress.
@johanneslenfers is actively working on improving the process.

I am sure that you have seen the explanation of the flies generated by the exploration process here: https://rise-lang.org/doc/exploration/tutorial#output-of-exploration-process

The `Random.csv` file contains an overview of the runtimes measured for each program evaluated during the exploration. It might be that for the example program the runtime results are not very exciting and generally small.

As far as I know, the `main` branch currently only supports the exploration of `C` programs, but @johanneslenfers is already exploring `OpenCL` programs as well. Maybe @johanneslenfers you can comment on that here.

We will definitely keep this discussion updated when we have improved the support in the `main` branch as well.

Best,
Michel
